addons/mail/models/mail_message.py

```diff
@@ -767,7 +767,8 @@ def base64_to_boundary(match):
                         'name': name,
                         'datas': match.group(2),
                         'datas_fname': name,
-                        'res_model': 'mail.message',
+                        'res_model': values.get('model'),
```

Comment:
Maybe: `'res_model': values.get('model', 'mail.message'),`

Reply:
If we do that we'll end up with an incorrect record; the message has not been created yet, so there is no res_id to populate the res_id field of the attachment.
If you really want to do that, then after line 780, once the message create has been done, we should add something along the lines of 'attachment.write({'res_id': message.id})'

I put a value.get('model') instead of value['model'] because of the tests, but I don't think it is really necessary otherwise.

Reply:
It seems good to me, is it possible to have no model and id? If so, I do not remember what becomes the attachment

Reply:
AFAIK, having a chatter without a res_model/res_id would mean it being without a parent object.
That looks like an anomalous situation to me, and I have no idea how that could happen.